### PR TITLE
Adding TMS support when user adds a XYZ basemap url

### LINKED ISSUE #3694: Add TMS option for XYZ URL add custom basemap modal
The TMS option indicate that the values of the Y-values should be inverted. It's an optional setting to be provided together with a XYZ url.

I proposed something on the lines of (since it was lacking in the mockup)
screen shot 2015-05-19 at 12 58 29
however, @saleiva rejected it with the motivation that you'd do a better one.. ;)

See #3685 for previous PR/context

### PR DESCRIPTION
**Pull request by xavijam**
- Add the possibility to support TMS in new basemap XYZ urls.
- Info tooltip (help for a better text please).
- Removed idle icon.

**CONS**

- Added a new css for this "view", impossible to manage new things without adding new styles, form-content is getting bigger and it doesn't make sense (unless we decide checkboxes will be placed at the end of the input from now on).

Fixes #3694
REVIEWER: @viddo

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3908/))

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3909/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3910/))

**Comment by xavijam** [comment]
retest this please

**Comment by xavijam** [comment]
@viddo, could you check it?
@josecruz, do you like the interaction?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3926/))

File: app/assets/stylesheets/editor/custom_basemap_dialog.css.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Custom styles for custom basemap styles
+// --------------------------------------------------
+
+
+.CustomBasemap {}
+
+// XYZ panel view
+.XYZPanel {}
+.Form-input--longer.XYZPanel-input {
```
Comment (viddo) [inline app/assets/stylesheets/editor/custom_basemap_dialog.css.scss]: I've said this in [earlier PRs](https://github.com/CartoDB/cartodb/pull/3700#discussion_r30679964), I really don't think it's a good idea to mix/override rules this way. If `.Form-input--longer` is not valid as it is, I'd much rather see `Form-input--longerSpecialFoobar` to keep the rules that affects that piece in one common place.
I think that would go for the other styles in this file.

File: lib/assets/javascripts/cartodb/common/dialogs/add_custom_basemap/add_custom_basemap_view.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ var TabsView = require('./tabs_view.js');
  */
 module.exports = BaseDialog.extend({
 
+  className: 'Dialog is-opening CustomBasemap',
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/add_custom_basemap/add_custom_basemap_view.js]: I think this line could be removed considering `CustomBasemap` rules doesn't actually contain any rules

**Comment by viddo** [comment]
Apart from the CSS thing :+1:

**Comment by josecruz** [comment]
Works perfect for me! Loving it :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3958/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3981/))

**Pull request merged**